lang/LangPrimSource/PyrPrimitive.cpp

```diff
@@ -390,6 +390,10 @@ int prFloat_AsStringPrec(struct VMGlobals *g, int numArgsPushed)
 	if (err) return err;
 
 	char fmt[8], str[256];
+	// if our precision is bigger than our stringsize, we can generate a very nasty buffer overflow here. So
+	if( precision <= 0 ) precision = 1;
+	if( precision >= 200 ) precision = 200; // Nothing is that big anyway. And we know we will be smaller than our 256 char string
```

Comment:
maybe increase the string size of `str` to something like 1024? double-precision floating point numbers have a reasonable range on the order of `10^308`